### PR TITLE
write: use ini parser to format credentials file

### PR DESCRIPTION
**Pull request by gord5500**
### Why

The `SharedCredentialsFile` has a good implementation for working with ini files but the output is that of `x=y`. The GO version of BMX had the output such that every `=` sign for each line was aligned. 

### How

Use a third party ini parser library that will let us update / create profiles. The libraries I looked through don't have this same feature for dotnet. 

I considered writing something custom to give us the same kind of output here. While I got that working, I was worried of edge cases where the time spent wouldn't be worth it. Last thing I want is to break someone's credentials file.

So the new output will be in the format `x = y` instead of `x=y`

### CODE REVIEW

File: src/D2L.Bmx/WriteHandler.cs [hunk 1]
```diff
@@ -18,6 +21,11 @@ BmxConfig config
 		_awsCreds = awsCreds;
 		_consolePrompter = consolePrompter;
 		_config = config;
+
+		var x = new IniParserConfiguration();
+
+
```
Comment (gord5500) [inline src/D2L.Bmx/WriteHandler.cs:27]:
```suggestion
```

**Comment by gord5500** [comment]
Oops forgot to open

File: src/D2L.Bmx/WriteHandler.cs [hunk 2]
```diff
@@ -18,6 +21,7 @@ BmxConfig config
 		_awsCreds = awsCreds;
 		_consolePrompter = consolePrompter;
 		_config = config;
+		_parser = new FileIniDataParser();
```
Comment (cfbao) [inline src/D2L.Bmx/WriteHandler.cs:24]: pleas pass it through the constructor parameter if only for consistency.

Comment (gord5500) [reply src/D2L.Bmx/WriteHandler.cs:24]: Is here where you mean? https://github.com/Brightspace/bmx/blob/main/src/D2L.Bmx/Program.cs#L131

Comment (cfbao) [reply src/D2L.Bmx/WriteHandler.cs:24]: yeah

Comment (cfbao) [reply src/D2L.Bmx/WriteHandler.cs:24]: Also, I think we can reuse this ini parser in `BmxConfigProvider` and get rid of `Microsoft.Extensions.Configuration.Ini`.
There are some nice things we can do with a proper parser+writer.
Another time though.

Comment (gord5500) [reply src/D2L.Bmx/WriteHandler.cs]: Thanks for clarification. I've moved it

File: src/D2L.Bmx/Program.cs [hunk 3]
```diff
@@ -131,6 +132,7 @@
 writeCommand.SetHandler( ( InvocationContext context ) => RunWithErrorHandlingAsync( context, () => {
 	var consolePrompter = new ConsolePrompter();
 	var config = new BmxConfigProvider().GetConfiguration();
+	var parser = new FileIniDataParser();
```
Comment (gord5500) [inline src/D2L.Bmx/Program.cs:135]: well maybe I should just put it directly in and not create a var. I was thinking maybe we want to configure it more before passing in but I don't think so now

**Review by cfbao — APPROVED** [review]

**Review by cfbao — APPROVED** [review]

**Pull request merged**